src/inference/gibbs.jl

```diff
@@ -36,25 +36,38 @@ end
 alg = Gibbs(HMC(0.2, 3, :v1), PG(20, :v2))
 ```
 
+One can also pass the number of iterations for each Gibbs component using the following syntax:
+- `alg = Gibbs((HMC(0.2, 3, :v1), n_hmc), (PG(20, :v2), n_pg))`
+- `alg = Gibbs(HMC(0.2, 3, :v1) => n_hmc, PG(20, :v2) => n_pg)`
+where `n_hmc` and `n_pg` are the number of iterations for HMC and PG in each Gibbs iteration.
+
 Tips:
 - `HMC` and `NUTS` are fast samplers, and can throw off particle-based
 methods like Particle Gibbs. You can increase the effectiveness of particle sampling by including
 more particles in the particle sampler.
 """
-struct Gibbs{space, A<:Tuple} <: InferenceAlgorithm
+struct Gibbs{space, A<:Tuple, B<:Tuple} <: InferenceAlgorithm
```

Comment:
Could even restrict it to

```suggestion
struct Gibbs{space,N, A<:NTuple{N,InferenceAlgorithm}, B<:NTuple{N,Int}} <: InferenceAlgorithm
```

BTW any change of the type parameters is in principle breaking as it can break downstream dispatches. So regardless of how the type parameters are changed, we should tag a breaking release.

Reply:
This looks more like a refactoring of `Gibbs` for me. Not sure if we want to do it in this PR.

Reply:
The PR is breaking in any case.

Even without `N` it would be good to ensure at some point that the lengths are equal and we could restrict type parameters to
```suggestion
struct Gibbs{space, A<:Tuple{Vararg{InferenceAlgorithm}}, B<:Tuple{Vararg{Int}}} <: InferenceAlgorithm
```